src/libaktualizr/uptane/ipsecondarydiscovery.cc

```diff
@@ -55,25 +57,32 @@ std::vector<Uptane::SecondaryConfig> IpSecondaryDiscovery::waitDevices() {
 
   while ((recieved = recvfrom(*socket_hdl, rbuf, sizeof(rbuf) - 1, 0, reinterpret_cast<struct sockaddr *>(&sec_address),
                               &sec_addr_len)) != -1) {
-    std::string data(rbuf, recieved);
-    try {
-      Uptane::SecondaryConfig conf;
-
-      int32_t sec_port;
-      asn1::Deserializer des(data);
-
-      des >> asn1::expl(AKT_DISCOVERY_RESP) >> asn1::seq >> asn1::implicit<kAsn1Utf8String>(conf.ecu_serial) >>
-          asn1::implicit<kAsn1Utf8String>(conf.ecu_hardware_id) >> asn1::implicit<kAsn1Integer>(sec_port) >>
-          asn1::restseq >> asn1::endexpl;
-
-      LOG_INFO << "Found secondary:" << conf.ecu_serial << " " << conf.ecu_hardware_id;
-      conf.secondary_type = Uptane::SecondaryType::kIpUptane;
-      conf.ip_addr = sec_address;
-      Utils::setSocketPort(&conf.ip_addr, htons(sec_port));
-      secondaries.push_back(conf);
-    } catch (const deserialization_error &ex) {
-      LOG_ERROR << ex.what();
+    Uptane::SecondaryConfig conf;
+    AKIpUptaneMes_t *m = nullptr;
+    asn_dec_rval_t res = ber_decode(nullptr, &asn_DEF_AKIpUptaneMes, reinterpret_cast<void **>(&m), rbuf, recieved);
+    Asn1Message::Ptr msg = Asn1Message::FromRaw(&m);
+
+    if (res.code != RC_OK) {
+      LOG_ERROR << "Failed to parse discovery response message";
```

Comment:
It looks like this function handles the case of incomplete messages gracefully (it will continue on `RC_WMORE`).

But in that case, it should probably not issue a parse error log, since it's a pretty normal case.

Reply:
The incoming messages are datagrams from the secondaries, so if they don't parse completely then I think the only think we can do is to consider `RC_WMORE` to be the same as `RC_FAIL`?

Was your question about how loudly we should complain on about parse errors?  I was unsure whether these should be warnings or errors. From a single application point of view, I'd consider this a warning: it shouldn't happen, but we can continue service correctly. However from a global point of view, there is probably something up: the secondary shouldn't be sending us junk like this.

Reply:
Oh ok yes, I forgot that this part uses datagram and made the comment thinking about tcp, sorry.

About the loudness, `LOG_ERROR` sounds about right then.